model-mesh-ovms-adapter/server/modelmanager.go

```diff
@@ -379,38 +306,136 @@ func (mm *OvmsModelManager) run() {
 				modelState = modelStatus.State
 			}
 
-			log.V(1).Info("Completing request", "model_id", id, "status_exists", statusExists, "type", req.requestType, "state", modelState)
+			var code codes.Code
+			var message string
+			if !statusExists {
+				code = codes.Internal
+				message = "Expected model to load, but no status entry found in the config"
+			} else if modelStatus.State == "AVAILABLE" {
+				code = codes.OK
+			} else {
+				code = codes.InvalidArgument
```

Comment:
Probably this should also be `Internal` unless we can map/interpret the error code from OVMS. `InvalidArgument` means that there was something wrong with the request itself (which we don't know is the case).

Reply:
In this case, I am interpreting `InvalidArgument` to mean that there is something wrong with the model data. The reload completed but the model could not be loaded. None of [the other status codes](https://grpc.github.io/grpc/core/md_doc_statuscodes.html) seem to cover that case.

The description for `Internal` is:
> This means that some invariants expected by the underlying system have been broken. This error code is reserved for serious errors.

Since it is expected that a model can fail to load, it doesn't seem right to call it `Internal`.

One place that the distinction may matter is if MM interprets different failure codes as retry-able or not. `Internal` is probably retry-able but `InvalidArgument` would not be. I'm not sure what ModelMesh does here though, so your call on which error is more appropriate.

On interpreting what OVMS returns in a failure case: In my testing of a few failure cases, the status returned for any failure is "LOADING" with an "UKNOWN" reason:
```
      {
        "version": "1",
        "state": "LOADING",
        "status": {
          "error_code": "UNKNOWN",
          "error_message": "UNKNOWN"
        }
      }
```

Reply:
I actually think `Unknown` is probably the best gRPC code to go with. Could possibly make anargument for `FailedPrecondition`.